vsphere/internal/vmworkflow/virtual_machine_customize_subresource.go

```diff
@@ -0,0 +1,490 @@
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+const (
+	cKeyPrefix        = "clone.0.customize.0"
+	cLinuxKeyPrefix   = "clone.0.customize.0.linux_options.0"
+	cWindowsKeyPrefix = "clone.0.customize.0.windows_options.0"
+	cNetifKeyPrefix   = "clone.0.customize.0.network_interface"
+)
+
+// netifKey renders a specific network_interface key for a specific resource
+// index.
+func netifKey(key string, n int) string {
+	return fmt.Sprintf("%s.%d.%s", cNetifKeyPrefix, n, key)
+}
+
+// matchGateway take an IP, mask, and gateway, and checks to see if the gateway
+// is reachable from the IP address.
+func matchGateway(a string, m int, g string) bool {
+	ip := net.ParseIP(a)
+	gw := net.ParseIP(g)
+	var mask net.IPMask
+	if ip.To4() != nil {
+		mask = net.CIDRMask(m, 32)
+	} else {
+		mask = net.CIDRMask(m, 128)
+	}
+	if ip.Mask(mask).Equal(gw.Mask(mask)) {
+		return true
+	}
+	return false
+}
+
+func v4CIDRMaskToDotted(mask int) string {
+	m := net.CIDRMask(mask, 32)
+	a := int(m[0])
+	b := int(m[1])
+	c := int(m[2])
+	d := int(m[3])
+	return fmt.Sprintf("%d.%d.%d.%d", a, b, c, d)
+}
+
+// VirtualMachineCustomizeSchema returns the schema for VM customization.
+func VirtualMachineCustomizeSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		// CustomizationGlobalIPSettings
+		"dns_server_list": {
+			Type:        schema.TypeList,
+			Optional:    true,
+			Description: "The list of DNS servers for a virtual network adapter with a static IP address.",
+			Elem:        &schema.Schema{Type: schema.TypeString},
+		},
+		"dns_suffix_list": {
+			Type:        schema.TypeList,
+			Optional:    true,
+			Description: "A list of DNS search domains to add to the DNS configuration on the virtual machine.",
+			Elem:        &schema.Schema{Type: schema.TypeString},
+		},
+
+		// CustomizationLinuxPrep
+		"linux_options": {
+			Type:          schema.TypeList,
+			Optional:      true,
+			MaxItems:      1,
+			ConflictsWith: []string{cKeyPrefix + "." + "windows_options", cKeyPrefix + "." + "windows_sysprep_text"},
+			Description:   "A list of configuration options specific to Linux virtual machines.",
+			Elem: &schema.Resource{Schema: map[string]*schema.Schema{
+				"domain": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "The FQDN for this virtual machine.",
+				},
+				"host_name": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "The host name for this virtual machine.",
+				},
+				"hw_clock_utc": {
+					Type:        schema.TypeBool,
+					Optional:    true,
+					Default:     true,
+					Description: "Specifies whether or not the hardware clock should be in UTC or not.",
+				},
+				"time_zone": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "Customize the time zone on the VM. This should be a time zone-style entry, like America/Los_Angeles.",
```

Comment:
Do we want to add any validation around this? I'm thinking just making sure it looks like `%s/%s`

Reply:
There's a lot of combinations here, and a slash is actually not necessary all the time it seems, but I've added validation anyway to make sure any extraneous characters that are not found in the TZ database can't be added.